### PR TITLE
Cache beatmap metadata lookups used by multiplayer

### PR DESCRIPTION
**Pull request by peppy**
The cache is a bonus really, mostly wanted the async queue logic (stolen from `UserLookupCache` as a tried-and-tested structure). This paves the way for further optimisations with these retrievals, but those optimisations are less important with this change.

https://github.com/ppy/osu-web/pull/8392 will allow for doing lookups of up to 50 beatmaps at once, similar to what the `UserLookupCache` component is using.

Hopefully this is okay without tests / abstract refactors. Don't want to sink too much time into this just yet as I'm not sure how often this will be used (or whether it will need to support `beatmap_set_id` as well, for instance).

### CODE REVIEW

File: osu.Game/Database/BeatmapLookupCache.cs
Comment (bdach) [inline osu.Game/Database/BeatmapLookupCache.cs:85]: These mentions of 50 are not correct as is, right now it's at most one beatmap. Not sure what the plan there is (leave this be until the web endpoint gets merged? roll forward with 1 for now?)

**Review by bdach — COMMENTED** [review]
I could be ok with this duplicated to avoid generalising too early, I think. Going to seek a second opinion anyhow.

File: osu.Game/Database/BeatmapLookupCache.cs
Comment (peppy) [reply osu.Game/Database/BeatmapLookupCache.cs:85]: Either or. I think the web change should roll out today so maybe best to wait on it.

Comment (peppy) [reply osu.Game/Database/BeatmapLookupCache.cs:85]: Has been updated to consume the new batch lookup request.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**
